### ISSUE TITLE
The web stuck on the rescheduling page

### ISSUE THREAD

**Issue by wenchendong** [kcajc/usvisa-ca#21]
The webpage is stuck on the rescheduling page and does not search for a date.
![image](https://github.com/user-attachments/assets/660e70d7-2ba0-436c-8126-2318bd07f355)

**Comment by kapahin** [comment]
Yes, I notice the same. I think you may need to get the date of appointment from the scheduling page rather than the payment page. In my case I'm first time booking the appointment (not rescheduling one)

**Comment by kcajc** [comment]
I might have fixed this in #24. Please try pulling the lasted and see if it's fixed.

Or it might be caused by the hardcoded URL [here](https://github.com/kcajc/usvisa-ca/blob/094149bd16a9a9fef8b86a57933f12cd451d2b51/settings.py#L41) (i.e. the Vancouver consulate might use a different number than "94"). Please use the developer tool in your browser to inspect the request sent on this page to see if this is the case. If you manage to fix it by changing the URL, please let me know.

**Comment by kcajc** [comment]
This project is only written and tested with RE-scheduling in mind, so booking your first appointment is beyond the scope of the project.

**Comment by wenchendong** [comment]
I just tried after you closed this. The issue still remains.

![Image](https://github.com/user-attachments/assets/fd5d8842-5e84-4b35-9cce-ce8b14fec847)

**Comment by kcajc** [comment]
I think there are users who are able to search for dates… Do you have an appointment booked already?

**Comment by wenchendong** [comment]
> I think there are users who are able to search for dates… Do you have an appointment booked already?

Yes, I have. I provided my account to you before through email for your maintenance. You can log in with my account to double check. Thanks!

**Comment by kcajc** [comment]
I have two midterms coming up this week🥲 Let me take a look after all these. Thanks for your help!

**Comment by wenchendong** [comment]
No worries. Good luck on your midterms and please take your time.

**Comment by kcajc** [comment]
@wenchendong I just checked and I noticed two potential reasons for the failure:

1. You seem to have a group appointment, this was not tested when I wrote the tool and might require some minor tweaking in the tool. Specifically, on the https://ais.usvisa-info.com/en-ca/niv/groups/<some number> page you probably need to either click on the "continue" button of an appointment or send a request that achieves the same effect.
2. You have an appointment for the Vancouver consulate, so you need to change `AVAILABLE_DATE_REQUEST_SUFFIX` in settings (see the updated README).

**Comment by wenchendong** [comment]
@kcajc  Thanks a lot for looking into this.
1. Emm, interesting. There is only one single person for that appointment. The other person has its own group and has gotten his US visa. I do not think that they are one group. Could you please double check?
2. Yes I did change the script accordingly but it still did not work.

**Comment by kcajc** [comment]
OK I think I misunderstood the problem, let me actually try with the code.

**Comment by kcajc** [comment]
@wenchendong The script is working (with your account):
<img width="520" alt="Image" src="https://github.com/user-attachments/assets/9f5b9e0f-330d-4c44-90d9-184bdbbba96c" />
If you just look at the terminal output it is trying to find slots, despite seemingly doing nothing in the Chrome interface.

**Comment by wenchendong** [comment]
Oh cool. You are abosolutely RIGHT! Sorry I ran THE python script through my Notepad++, which did not present the running message. The reschedule.py works now. It seems that "detect_and_notify.py" still not but it is good enough. Thank you very much!

**Comment by kcajc** [comment]
No problem. And yes, `detect_and_notify.py` is broken for now.
